UndertaleModTool/MainWindow.xaml.cs

```diff
@@ -2148,9 +2177,200 @@ private void MenuItem_Add_Click(object sender, RoutedEventArgs e)
             OpenInTab(obj, true);
         }
 
-        private void RootMenuItem_SubmenuOpened(object sender, RoutedEventArgs e)
+        private void MenuItem_CollapseAll_Click(object sender, RoutedEventArgs e)
         {
-            MenuItem_RunScript_SubmenuOpened(sender, e, Path.Combine(ExePath, "Scripts"));
+            foreach (TreeViewItem item in (MainTree.Items[0] as TreeViewItem).Items)
+                item.IsExpanded = false;
+        }
+
+        private void MenuItem_CollapseCurrent_Click(object sender, RoutedEventArgs e)
+        {
+            foreach (TreeViewItem item in (MainTree.Items[0] as TreeViewItem).Items)
+                if (item.Items.Contains(MainTree.SelectedItem))
+                    item.IsExpanded = false;
+        }
+
+        private void MenuItem_ContextMenuOpened(object sender, RoutedEventArgs e)
```

Comment:
please double check codestyle in this whole function, not gonna point it out in every instance.